apps/contrib/templatetags/force_translation.py

```diff
@@ -0,0 +1,33 @@
+from django.template import Library, Node, TemplateSyntaxError
+from django.utils import translation
+
+register = Library()
+
+
+class TransNode(Node):
+    def __init__(self, nodelist, lc):
+        self.nodelist = nodelist
+        self.lc = lc
+
+    def render(self, context):
+        with translation.override(self.lc):
+            return self.nodelist.render(context)
+
+
+def force_translation(parser, token):
```

Comment:
I just kind of understand what is happening in this function. Can you provide some explanation?

Reply:
Basically it is a parser for the force_translation template tag parser. It will be called if the django template engine meets a `{% force_language %}` tag, due to its registration below. Then it will parse until `{% endforce_translation %}` and discard the the `{% endforce_translation %}` tag from the parser. Finally there is some validation of the template tag arguments.